### ISSUE TITLE
use HEX color values in addition to RGB

### ISSUE THREAD

**Issue by evdama** [jonaskay/tailwindcss-elevation#4]
Hi,

this package is brilliant, I like it for the fact that it gets me that bit of material design I actually like, not the whole shebang I don't need.

However, I've set it up and it's performing nicely. I do have two themes though, light and dark which I switch between using CSS variables. See gists here https://gist.github.com/evdama/6a3ed83f4528b5f9ad5d7743c1646096 and https://gist.github.com/e8c5540d7ea1a6b8e6792edf31009ffd

I'm using the colors from the `colors` object inside the tw conf file as you can see, grab them into the two theme objects using tailwinds `theme()` and then onward to wherever stuff needs to go inside the app.

As you can see from line 345 in tailwind conf file I'm using a CSS variables that injects a different hex color value based on which theme is currently choosen by the user. So what's needed is for elevation to not just accept RGB values but also hex.

I thought about doing a custom local fork with a code snippet from e.g. https://css-tricks.com/converting-color-spaces-in-javascript/ but then I though it would be much better also for others if this would be added upstream.

**Comment by jonaskay** [comment]
Hi @evdama,

Thank you for the kind words and making the effort to write this issue.

I think having the option to submit the color config as a HEX value makes sense. It's annoying having to convert RGB to HEX.

Just to make sure that we understand each other: the idea is that the following two configs would result in the same CSS output.

```
module.exports = {
  plugins: [
    require('tailwindcss-elevation')(
      [],
      {
        color: '77,192,181',
        opacityBoost: '0.23'
      }
    )
  ]
}
```

```
module.exports = {
  plugins: [
    require('tailwindcss-elevation')(
      [],
      {
        color: '#4dc0b5',
        opacityBoost: '0.23'
      }
    )
  ]
}
```

**Comment by evdama** [comment]
Yes, that's what I mean :)

**Comment by evdama** [comment]
I'd use a CSS variable as shown here https://gist.github.com/evdama/e8c5540d7ea1a6b8e6792edf31009ffd#file-tailwind-config-js-L367 but that is just individual config atop the fact that you can provide a hex value as well

**Comment by jonaskay** [comment]
This feature (https://github.com/jonaskay/tailwindcss-elevation/pull/6) is now available in version 0.3.2.

**Comment by evdama** [comment]
excellent, it works with hex values now... one last thing I need to figure out is how to make it work with CSS variables as in this code, setting the hex value directly is ok, but setting it with the CSS variable gives `Invalid color value: var(--theme-color-bg-four)`

```
plugins: [
    require('tailwindcss-elevation')(['responsive, hover, focus'], {
      // color: shadowColor,
      color: 'var(--theme-color-bg-four)',
      // color: '#AEEA00',
      opacityBoost: '0.1',
    }),
    require('@tailwindcss/custom-forms'),
  ]
```

**Comment by jonaskay** [comment]
Those CSS variables got caught by the config validator. This issue is fixed here https://github.com/jonaskay/tailwindcss-elevation/pull/7. Fix is released with 0.3.3.

Just as a warning: **using a HEX triplet as a CSS variable won't work.** You have to use an RGB triplet. More about this here: https://stackoverflow.com/questions/29591465/use-css-variables-with-rgba-for-gradient-transparency
